mobile/library/java/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java

```diff
@@ -159,6 +164,11 @@ public EnvoyConfiguration(
     this.enableDrainPostDnsRefresh = enableDrainPostDnsRefresh;
     this.enableHttp3 = enableHttp3;
     this.useCares = useCares;
+    this.fallbackResolvers = new ArrayList<>();
+    for (Pair<String, Integer> hostAndPort : fallbackResolvers) {
```

Comment:
It's a bit wasteful to convert from `List<Pair<String, Integer>` to `List<Pair<String, String>` and then `byte[][]`. Can we convert from `List<Pair<String, Integer>>` to `byte[][]` right away?

Reply:
Yeah we use this toString pattern in multiple places and it's probably not the best solution.

We can probably follow up to clean this up. What alternatives are you thinking? Using a Bytebuffer to convert the integer?

Reply:
Yeah, we can do this in a follow-up.